lib/collection/bloc/collection_bloc.dart

```diff
@@ -527,72 +607,31 @@ class CollectionBloc extends Bloc<CollectionEvent, CollectionState> {
         geoLocationStatus: GeoLocationStatus.none));
   }
 
-  Future<void> _onGeoLocationRequested(
-      GeoLocationRequested event, Emitter<CollectionState> emit) async {
-    bool serviceEnabled;
-    LocationPermission permission;
+  void _onGibsonsFormDuplicated(
+      GibsonsFormDuplicated event, Emitter<CollectionState> emit) async {
```

Comment:
Now that I think about it, it is `async` because of the `await` call to the `Future` returning function. But that seems to be also callable without the `await`, but I'm not sure what will that result it, and it still wouldn't solve the concurrency bug.